cachalot/tests/read.py

```diff
@@ -506,17 +516,22 @@ def test_filtered_relation(self):
         qs = TestChild.objects.annotate(
             filtered_permissions=FilteredRelation(
                 'permissions', condition=Q(permissions__pk__gt=1)))
-        self.assert_tables(qs, TestChild)
+        # Resulting query:
+        # SELECT "cachalot_testparent"."id", "cachalot_testparent"."name",
+        # "cachalot_testchild"."testparent_ptr_id", "cachalot_testchild"."public"
+        # FROM "cachalot_testchild" INNER JOIN "cachalot_testparent" ON
+        # ("cachalot_testchild"."testparent_ptr_id" = "cachalot_testparent"."id")
+        self.assert_tables(qs, TestParent, TestChild)
```

Comment:
I thought about this for a while. It seems to be correct as well because in Django it’s valid to lookup an object of this model and do operations on it: `TestParent.objects.all().first().update(...)`

Reply:
`TestParent.objects.all().first().update(...)` is two queries. `.first()` performs a slice. If you did `.filter()`, you'd be doing an `UPDATE`.

Reply:
> ~`TestParent.objects.all().first().update(...)` is two queries. `.first()` performs a slice. If you did `.filter()`, you'd be doing an `UPDATE`.~ Actually, would it? Or does the slice just do a `LIMIT`?

Let me clarify the example:
```
cached_query = TestChild.objects.all()
# Doing an update on the parent objects
TestParent.objects.update(name="something else")
# Does not invalidate cached_query because TestChild is not involved?
```

Reply:
I believe that is the correct behavior. It's not like the M2M table to Permissions changed since you didn't delete TestParent, thus there's no cascading signal that Django performs right?

Reply:
Ok nvm I see TestChild has an auto created OneToOneField.